tendrl/common/singleton.py

```diff
@@ -0,0 +1,8 @@
+def singleton(cls):
```

Comment:
Please change method name to something more meaningful, eg: to_singleton, make_singleton